executor/join_result_generators.go

```diff
@@ -315,14 +571,49 @@ func (outputer *leftOuterJoinResultGenerator) emitMatchedInners(outer Row, inner
 		buffer = outputer.makeJoinRowToBuffer(buffer[len(buffer):], outer, inner)
 		resultBuffer = append(resultBuffer, buffer)
 	}
-	return outputer.filterResult(resultBuffer, originLen)
+	var matched bool
+	var err error
+	resultBuffer, matched, err = outputer.filterResult(resultBuffer, originLen)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !matched {
+		resultBuffer = outputer.emitUnMatchedOuter(outer, resultBuffer)
+	}
+	return resultBuffer, nil
+}
+
+// emitMatchedInnersToChunk implements joinResultGenerator interface.
+func (outputer *leftOuterJoinResultGenerator) emitMatchedInnersToChunk(outer chunk.Row, inners []chunk.Row, chk *chunk.Chunk) error {
+	if len(inners) == 0 {
+		outputer.emitUnMatchedOuterToChunk(outer, chk)
+		return nil
+	}
+	outputer.chk.Reset()
+	for _, inner := range inners {
+		outputer.makeJoinRowToChunk(outputer.chk, outer, inner)
```

Comment:
If we don't have a filter, we can make join row directly to `chk` instead of `outputer.chk`.